File: src/ElasticElmah.Appender/Storage/LogEvent.cs
using System.Collections.Generic;

namespace ElasticElmah.Appender.Storage
{
    public class LogEvent
    {
        public LogEvent()
        {
            locationInfo = new LogEventLocation();
        }
        public string loggerName { get; set; }
        public string level { get; set; }
        public string message { get; set; }
        public string threadName { get; set; }
        public string timeStamp { get; set; }
        public LogEventLocation locationInfo { get; set; }
        public string userName { get; set; }
        public Dictionary<string, object> properties { get; set; }
        public string exceptionString { get; set; }
        public string domain { get; set; }
        public string identity { get; set; }
    }
}

File: src/ElasticElmah.Append/LogEvent.fs
namespace ElasticElmah.Append
open System
open System.Collections.Generic
type LogEventLocation=
    { className :string;
      fileName :string;
      lineNumber :string;
      methodName :string;
    }
    
type LoggingEvent=
    {
        loggerName :string;
        level :string;
        message :string;
        threadName :string;
        timeStamp :string;
        locationInfo :LogEventLocation;
        userName :string;
        properties :Dictionary<string,Object>;
        exceptionString :string;
        domain :string;
        identity :string;
    }
